pkg/flare/writer.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package flare
+
+import (
+	"bufio"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	log "github.com/cihub/seelog"
+)
+
+//RedactingWriter is a writer that will redact content before writing to target
+type RedactingWriter struct {
+	target    *os.File
+	targetBuf *bufio.Writer
+	perm      os.FileMode
+	r         []replacer
+}
+
+//NewRedactingWriter instantiates a RedactingWriter to target with given permissions
+func NewRedactingWriter(t string, p os.FileMode, buffered bool) (*RedactingWriter, error) {
+	err := ensureParentDirsExist(t)
```

Comment:
do we want to create this directory automatically ?

Reply:
we did want to, but it's probably weird to have it in the `New()`, so I'm moving that logic out to the flare creating routines.